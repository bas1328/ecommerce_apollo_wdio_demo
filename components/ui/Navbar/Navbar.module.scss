.container {
    display: flex;
    justify-content: center;
    gap: 5rem;
    margin: 5rem;
    font-size: 1.6rem;
}