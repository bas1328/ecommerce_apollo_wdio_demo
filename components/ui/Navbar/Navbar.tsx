import React from "react";
import styles from "./Navbar.module.scss";
import Link from "next/link";

export default function Navbar() {
  return (
    <div className={styles.container}>
      <Link href="/">home 🏠</Link>
      <Link href="/product-list">our products 👕</Link>
    </div>
  );
}
