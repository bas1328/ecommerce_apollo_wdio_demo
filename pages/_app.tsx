import { ApolloProvider } from "@apollo/client";
import createApolloClient from "../apollo-client";
import MainNav from "@/components/layouts/MainNav/MainNav";
import "@/styles/globals.css";
import type { AppProps } from "next/app";

export default function App({ Component, pageProps }: AppProps) {
  const client = createApolloClient();

  return (
    <ApolloProvider client={client}>
      <MainNav>
        <Component {...pageProps} />
      </MainNav>
    </ApolloProvider>
  );
}
