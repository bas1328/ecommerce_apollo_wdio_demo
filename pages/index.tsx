import Head from "next/head";
import Image from "next/image";
import { Inter } from "next/font/google";
import styles from "@/styles/Home.module.css";

const inter = Inter({ subsets: ["latin"] });

export default function Home() {
  return (
    <>
      <Head>
        <title>demo shop</title>
        <meta name="description" content="demo ecommerce shop" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <main className={`${styles.main} ${inter.className}`}>
        <h1>Lorem ipsum dolor sit amet consectetur adipisicing elit. Ex, aspernatur!</h1>
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Perspiciatis tempora deserunt quia nostrum temporibus ipsa. Esse libero animi quo ex facilis sequi vitae, reprehenderit quasi officiis officia ducimus nesciunt rerum?</p>
      </main>
    </>
  );
}
