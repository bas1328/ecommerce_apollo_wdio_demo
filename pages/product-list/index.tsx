import Link from "next/link";
import createApolloClient from "@/apollo-client";
import { GET_ALL_PRODUCTS_QUERY } from "@/queries/getAllProductsQuery";

import ProductItem from "@/pages/product-list/components/ProductItem/ProductItem";

import styles from "./ProductList.module.scss";

// TODO: add types
export default function ProductList({ characters }: any) {
  return (
    <main className={styles.main}>
      <h1>Our products</h1>
      {Boolean(characters) ? (
        <>
          <ul>
            {characters?.map((product: any) => (
              <li key={product.id}>
                <ProductItem
                  name={product.name}
                  image={product.image}
                  id={product.id}
                />
              </li>
            ))}
          </ul>
        </>
      ) : null}
    </main>
  );
}

export async function getServerSideProps() {
  const client = createApolloClient();

  const { data } = await client.query({
    query: GET_ALL_PRODUCTS_QUERY,
  });

  return {
    props: {
      characters: data.characters.results,
    },
  };
}
