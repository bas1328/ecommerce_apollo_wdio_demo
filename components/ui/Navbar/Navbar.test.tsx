import { render } from "@testing-library/react";
import Navbar from "./Navbar";
import "@testing-library/jest-dom";
import { BasketContext } from "@/pages/_app";
import { Character } from "@/gql/graphql";

const basketItems = [] as Character[];
const setBasketItems = jest.fn();

describe("Navbar", () => {
  it("renders a Navbar", () => {
    const screen = render(
      <BasketContext.Provider
        value={{
          basketItems,
          setBasketItems,
        }}
      >
        <Navbar />
      </BasketContext.Provider>
    );

    const heading = screen.getByRole("link", {
      name: /home/i,
    });

    expect(heading).toBeInTheDocument();
    expect(screen).toMatchSnapshot();
  });
});
