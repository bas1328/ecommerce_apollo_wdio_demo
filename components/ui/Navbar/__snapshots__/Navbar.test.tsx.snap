// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders a Navbar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container className"
      >
        <div>
          <a
            href="/"
          >
            home |
          </a>
          <a
            href="/product-list"
          >
            | our t-shirts
          </a>
        </div>
        <button
          aria-controls="radix-:r0:"
          aria-expanded="false"
          aria-haspopup="dialog"
          data-state="closed"
          id="basket"
          type="button"
        >
          <img
            alt="basket"
            data-nimg="1"
            decoding="async"
            height="30"
            loading="lazy"
            src="/_next/image?url=%2Fbasket.png&w=64&q=75"
            srcset="/_next/image?url=%2Fbasket.png&w=32&q=75 1x, /_next/image?url=%2Fbasket.png&w=64&q=75 2x"
            style="color: transparent;"
            width="30"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container className"
    >
      <div>
        <a
          href="/"
        >
          home |
        </a>
        <a
          href="/product-list"
        >
          | our t-shirts
        </a>
      </div>
      <button
        aria-controls="radix-:r0:"
        aria-expanded="false"
        aria-haspopup="dialog"
        data-state="closed"
        id="basket"
        type="button"
      >
        <img
          alt="basket"
          data-nimg="1"
          decoding="async"
          height="30"
          loading="lazy"
          src="/_next/image?url=%2Fbasket.png&w=64&q=75"
          srcset="/_next/image?url=%2Fbasket.png&w=32&q=75 1x, /_next/image?url=%2Fbasket.png&w=64&q=75 2x"
          style="color: transparent;"
          width="30"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
